pkg/dogstatsd/listeners/packet_buffer.go

```diff
@@ -0,0 +1,61 @@
+package listeners
+
+import (
+	"sync"
+	"time"
+)
+
+type packetBuffer struct {
```

Comment:
I could see this being useful elsewhere. Maybe not necessarily something to address in this PR, but it might make sense to refactor this at some point, export, and maybe set in its own relevant package or whatever. Nice one.

Reply:
Yup, I was feeling the same. Let's refactor when the need emerge.